### PR TITLE
watchdog: add watchdog driver support to mimxrt685 platform

### PR DESCRIPTION
**Pull request by dleach02**
Add watchdog support to the mimxrt685 platform.

The mimxrt685 platform is excluded from the watchdog
test case because the test case uses variables in the
noinit section that need to be retained through a reset
but the rt685 does not retain this memory through a
reset.

Signed-off-by: David Leach <david.leach@nxp.com>

### CODE REVIEW

**Review by CristXu — APPROVED** [review]

**Review by hakehuang — APPROVED** [review]

File: drivers/watchdog/wdt_mcux_wwdt.c [hunk 1]
```diff
@@ -81,8 +80,15 @@ static int mcux_wwdt_install_timeout(const struct device *dev,
 		return -ENOMEM;
 	}
 
+#if defined(CONFIG_SOC_MIMXRT685S_CM33)
+	/*CLOCK_SetClkDiv(kCLOCK_DivWdtClk, config->clk_divider);*/
```
Comment (MaureenHelm) [inline drivers/watchdog/wdt_mcux_wwdt.c:84]: Remove this line

File: west.yml [hunk 2]
```diff
@@ -92,7 +92,7 @@ manifest:
       groups:
         - hal
     - name: hal_nxp
-      revision: 4b493d4346e841e786a8daa2eb2ee03cbe5cfd37
+      revision: pull/102/head
```
Comment (MaureenHelm) [inline west.yml:95]: The hal PR has now been merged, so you can update this to the new SHA

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: drivers/watchdog/wdt_mcux_wwdt.c (hunk 1 above)
Comment (dleach02) [reply drivers/watchdog/wdt_mcux_wwdt.c]: done

File: west.yml (hunk 2 above)
Comment (dleach02) [reply west.yml]: done

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**